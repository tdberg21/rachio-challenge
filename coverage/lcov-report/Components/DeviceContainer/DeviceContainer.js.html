<!doctype html>
<html lang="en">
<head>
    <title>Code coverage report for Components/DeviceContainer/DeviceContainer.js</title>
    <meta charset="utf-8" />
    <link rel="stylesheet" href="../../prettify.css" />
    <link rel="stylesheet" href="../../base.css" />
    <meta name="viewport" content="width=device-width, initial-scale=1">
    <style type='text/css'>
        .coverage-summary .sorter {
            background-image: url(../../sort-arrow-sprite.png);
        }
    </style>
</head>
<body>
<div class='wrapper'>
  <div class='pad1'>
    <h1>
      <a href="../../index.html">All files</a> / <a href="index.html">Components/DeviceContainer</a> DeviceContainer.js
    </h1>
    <div class='clearfix'>
      <div class='fl pad1y space-right2'>
        <span class="strong">100% </span>
        <span class="quiet">Statements</span>
        <span class='fraction'>7/7</span>
      </div>
      <div class='fl pad1y space-right2'>
        <span class="strong">100% </span>
        <span class="quiet">Branches</span>
        <span class='fraction'>2/2</span>
      </div>
      <div class='fl pad1y space-right2'>
        <span class="strong">100% </span>
        <span class="quiet">Functions</span>
        <span class='fraction'>2/2</span>
      </div>
      <div class='fl pad1y space-right2'>
        <span class="strong">100% </span>
        <span class="quiet">Lines</span>
        <span class='fraction'>7/7</span>
      </div>
    </div>
  </div>
  <div class='status-line high'></div>
<pre><table class="coverage">
<tr><td class="line-count quiet">1
2
3
4
5
6
7
8
9
10
11
12
13
14
15
16
17
18
19
20
21
22
23
24
25
26
27
28
29</td><td class="line-coverage quiet"><span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-yes">2x</span>
<span class="cline-any cline-yes">2x</span>
<span class="cline-any cline-yes">1x</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-yes">1x</span>
<span class="cline-any cline-yes">2x</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-yes">1x</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-yes">2x</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span></td><td class="text"><pre class="prettyprint lang-js">import React from 'react';
import Device from '../Device/Device.js';
import './DeviceContainer.css';
import PropTypes from 'prop-types';
&nbsp;
const DeviceContainer = ({ devices }) =&gt; {
  if (!devices) {
    return (
      &lt;div className='device-container-component loading-container'&gt;
        &lt;p className='loading-text'&gt;Loading...&lt;/p&gt;
      &lt;/div&gt;
    );
  } else {
    let devicesToDisplay = devices.map((device, index) =&gt; {
      return &lt;Device device={device} key={index}/&gt;;
    });
    return (
      &lt;div className='device-container-component'&gt;
        {devicesToDisplay}
      &lt;/div&gt;
    );
  }
};
&nbsp;
export default DeviceContainer;
&nbsp;
DeviceContainer.propTypes = {
  devices: PropTypes.array
};</pre></td></tr>
</table></pre>
<div class='push'></div><!-- for sticky footer -->
</div><!-- /wrapper -->
<div class='footer quiet pad2 space-top1 center small'>
  Code coverage
  generated by <a href="https://istanbul.js.org/" target="_blank">istanbul</a> at Thu Oct 04 2018 19:19:04 GMT-0600 (MDT)
</div>
</div>
<script src="../../prettify.js"></script>
<script>
window.onload = function () {
        if (typeof prettyPrint === 'function') {
            prettyPrint();
        }
};
</script>
<script src="../../sorter.js"></script>
</body>
</html>
