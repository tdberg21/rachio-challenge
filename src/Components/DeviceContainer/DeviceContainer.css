
.loading-container {
  width: 30%;
  margin: 1em auto;
  text-align: center;
}

.loading-text {
  font-size: 1.2em;
}

.device-container-component {
  display: flex;
  flex-direction: column;
}

.device-card-container {
  display: flex;
  flex-direction: column;
  width: 70%;
  margin: 1em auto;
  background-color: rgba(255, 255, 255, 0.8);
  padding: 1rem;
  box-shadow: 2px 2px 2px 1px rgba(0, 0, 0, 0.2), -2px 2px 1px rgba(0, 0, 0, 0.2);
}

.device-card {
  display: flex;
}

.device-card-text-section {
  padding-left: 25px;
  width: 50%;
}

.device-card-input-section {
  display: flex;
  flex-direction: column;
  padding-top: 10px;
  width: 50%;
}

.duration-input {
  width: 64px;
  margin: 5px auto;
}

.duration-label {
  margin: 5px auto;
}

.activate-buttons {
  background-color: #3CA710;
}

.display-zones-button {
  background-color: #00a7e1;
}