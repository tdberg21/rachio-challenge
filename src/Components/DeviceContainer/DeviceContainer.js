import React from 'react';
import Device from '../Device/Device.js';
import './DeviceContainer.css';
import PropTypes from 'prop-types';

const DeviceContainer = ({ devices }) => {
  if (!devices) {
    return (
      <div className='device-container-component loading-container'>
        <p className='loading-text'>Loading...</p>
      </div>
    );
  } else {
    let devicesToDisplay = devices.map((device, index) => {
      return <Device device={device} key={index}/>;
    });
    return (
      <div className='device-container-component'>
        {devicesToDisplay}
      </div>
    );
  }
};

export default DeviceContainer;

DeviceContainer.propTypes = {
  devices: PropTypes.array
};