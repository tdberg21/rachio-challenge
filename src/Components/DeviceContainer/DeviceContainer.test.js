import React from 'react';
import Enzyme, { shallow } from 'enzyme';
import DeviceContainer from './DeviceContainer.js';
import Adapter from 'enzyme-adapter-react-16';
Enzyme.configure({ adapter: new Adapter() });

describe('DeviceContainer tests', () => {

  it('should match the snapshot without devices', () => {
    const wrapper = shallow(<DeviceContainer />);
    expect(wrapper).toMatchSnapshot();
  });

  it('should match the snapshot with devices', () => {
    let mockDevices = [{ id: 1 }, { id: 2 }];
    const wrapper = shallow(<DeviceContainer devices={mockDevices} />);
    expect(wrapper).toMatchSnapshot();
  });
});