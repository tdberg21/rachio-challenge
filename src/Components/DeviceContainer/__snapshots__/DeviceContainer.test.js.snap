// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`DeviceContainer tests should match the snapshot with devices 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <DeviceContainer
    devices={
      Array [
        Object {
          "id": 1,
        },
        Object {
          "id": 2,
        },
      ]
    }
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <Device
          device={
            Object {
              "id": 1,
            }
          }
        />,
        <Device
          device={
            Object {
              "id": 2,
            }
          }
        />,
      ],
      "className": "device-container-component",
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": "0",
        "nodeType": "class",
        "props": Object {
          "device": Object {
            "id": 1,
          },
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": "1",
        "nodeType": "class",
        "props": Object {
          "device": Object {
            "id": 2,
          },
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <Device
            device={
              Object {
                "id": 1,
              }
            }
          />,
          <Device
            device={
              Object {
                "id": 2,
              }
            }
          />,
        ],
        "className": "device-container-component",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": "0",
          "nodeType": "class",
          "props": Object {
            "device": Object {
              "id": 1,
            },
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": "1",
          "nodeType": "class",
          "props": Object {
            "device": Object {
              "id": 2,
            },
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
    "attachTo": undefined,
    "hydrateIn": undefined,
  },
}
`;

exports[`DeviceContainer tests should match the snapshot without devices 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <DeviceContainer />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": <p
        className="loading-text"
      >
        Loading...
      </p>,
      "className": "device-container-component loading-container",
    },
    "ref": null,
    "rendered": Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": "Loading...",
        "className": "loading-text",
      },
      "ref": null,
      "rendered": "Loading...",
      "type": "p",
    },
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": <p
          className="loading-text"
        >
          Loading...
        </p>,
        "className": "device-container-component loading-container",
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": "Loading...",
          "className": "loading-text",
        },
        "ref": null,
        "rendered": "Loading...",
        "type": "p",
      },
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
    "attachTo": undefined,
    "hydrateIn": undefined,
  },
}
`;
